templates/components/faceted-search/show-more-facets.html

```diff
@@ -1,6 +1,10 @@
 <div class="modal-body">
     <h1 class="facet-quick-heading">
-        {{all_facets}}
+        {{#if facet_title}}
+            {{facet_title}}
+        {{else}}
+            {{all_facets}}
+        {{/if}}
```

Comment:
This will check if `facet_title` exists and is not `null`, `undefined`, or `false`. If it exists, the Display Name is shown; otherwise, a fallback to `all_facets` as was before.